manager/src/main/java/com/redhat/service/bridge/manager/workers/resources/AbstractWorker.java

```diff
@@ -0,0 +1,181 @@
+package com.redhat.service.bridge.manager.workers.resources;
+
+import java.time.ZonedDateTime;
+import java.util.Objects;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.bridge.infra.models.dto.ManagedResourceStatus;
+import com.redhat.service.bridge.manager.models.ManagedResource;
+import com.redhat.service.bridge.manager.workers.Work;
+import com.redhat.service.bridge.manager.workers.WorkManagerImpl;
+import com.redhat.service.bridge.manager.workers.Worker;
+
+import io.quarkus.hibernate.orm.panache.PanacheRepositoryBase;
+
+public abstract class AbstractWorker<T extends ManagedResource> implements Worker<T> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractWorker.class);
+
+    @ConfigProperty(name = "event-bridge.resources.worker.max-retries")
+    int maxRetries;
+
+    @ConfigProperty(name = "event-bridge.resources.workers.timeout-seconds")
+    int timeoutSeconds;
+
+    @Inject
+    WorkManagerImpl workManager;
+
+    @Override
+    public void handleWork(Work work) {
+        //By the time VertX gets to execute an item of Work it may have been deleted.
+        if (!workManager.exists(work)) {
+            return;
+        }
+
+        T managedResource = load(work);
+        if (Objects.isNull(managedResource)) {
+            //Work has been scheduled but cannot be found. Something (horribly) wrong has happened.
+            String message = String.format("Resource of type '%s' with id '%s' no longer exists in the database.", work.getType(), work.getManagedResourceId());
+            throw new IllegalStateException(message);
+        }
+
+        boolean complete = false;
+        if (managedResource.getStatus() == ManagedResourceStatus.ACCEPTED) {
+            complete = createDependencies(managedResource).getDependencyStatus().isReady();
+        } else if (managedResource.getStatus() == ManagedResourceStatus.DEPROVISION) {
+            complete = deleteDependencies(managedResource).getDependencyStatus().isDeleted();
+        }
+
+        if (complete) {
+            workManager.complete(work);
+        } else {
+            workManager.schedule(managedResource);
+        }
+    }
+
+    @Transactional
+    protected T load(Work work) {
+        return getDao().findById(work.getManagedResourceId());
+    }
+
+    @Transactional
+    protected T setStatus(T managedResource, ManagedResourceStatus status) {
```

Comment:
The `ManagedResource` passed around is disconnected from the `EntityManager` and `Session` to keep transactions as short as possible. Therefore, whenever I need to update something, I need to reattach the `ManagedResource`, make the change and persist. Furthermore the reads on _lazy_ fields needed to be added into a transaction.